backend/publicKey/create.ts

```diff
@@ -0,0 +1,152 @@
+import { generateCreationResponse, PublicKeyAlgorithm } from '../authenticator'
+import type { CreateAuthenticatorOptions } from '../index'
+import { checkUserVerification, filterCredentials } from '../util'
+import type { CollectedClientData } from '../../types/interface'
+
+/**
+ *
+ * https://w3c.github.io/webauthn/#sctn-createCredential
+ */
+export async function create(
+    createOptions: CreateAuthenticatorOptions,
+    options: PublicKeyCredentialCreationOptions,
+    signal?: AbortSignal,
+): Promise<PublicKeyCredential | null> {
+    // we dont trust these parameters from upstream
+    delete options.timeout
+    delete options.rp.id
+
+    const { rpID, ...normalizedOptions } = await createOptions.getNormalizedCreateOptions()
+    if (!(await createOptions.hasCredential(rpID))) {
+        return new Promise((resolve, reject) => {
+            reject(new DOMException('NotSupportedError'))
+        })
+    }
```

Comment:
You should handle residential key and non-residential key differently

Reply:
What's the difference?